src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,106 @@
+import java.util.Optional;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+import mate.academy.util.ConnectionUtil;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    @Override
+    public Book create(Book book) {
+        String query = "INSERT INTO books (title, price) "
+                + "VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement
+                         = connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                book.setId(resultSet.getObject(1, Long.class));
+            }
+            return book;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't create book: " + book, e);
+        }
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        String query = "SELECT id, title, price FROM books WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            Book book = null;
+            if (resultSet.next()) {
+                book = getBook(resultSet);
+            }
+            return Optional.ofNullable(book);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get book by id: " + id, e);
+        }
+    }
+
+    @Override
+    public List<Book> findAll() {
+        String query = "SELECT id, title, price FROM books;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(query)) {
+            List<Book> books = new ArrayList<>();
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                books.add(getBook(resultSet));
+            }
+            return books;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get a list of books "
+                    + "from books table.", e);
+        }
+    }
+
+    @Override
+    public Book update(Book book) {
+        String query = "UPDATE books SET title = ?, price = ? WHERE id= ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            statement.setLong(3, book.getId());
+            return book;
```

Comment:
You are not closing the `PreparedStatement` in the `update` method. Use try-with-resources or explicitly close the statement to avoid potential resource leaks.